tests/models/test_configuration.py

```diff
@@ -300,6 +300,53 @@ def test_collection_mirror_settings(self):
             ExternalIntegrationLink.NO_MIRROR_INTEGRATION)
         eq_(settings[1]["options"][0]['label'],
             _("None - Do not mirror free books"))
+    
+    def test_relationships(self):
+        # Create a collection with two storage external integrations.
+        collection = self._collection(
+            name="Collection", protocol=ExternalIntegration.OVERDRIVE,
+        )
+
+        storage1 = self._external_integration(
+            name="integration1",
+            protocol=ExternalIntegration.S3,
+        )
+        storage2 = self._external_integration(
+            name="integration2",
+            protocol=ExternalIntegration.S3,
+            goal=ExternalIntegration.STORAGE_GOAL,
+            username="username", password="password",
+        )
+
+        # Two external integration links need to be created to associate
+        # the collection's external integration with the two storage
+        # external integrations.
+        s1_external_integration_link = self._external_integration_link(
+            integration=collection.external_integration,
+            other_integration=storage1, purpose="covers"
+        )
+        s2_external_integration_link = self._external_integration_link(
+            integration=collection.external_integration,
+            other_integration=storage2, purpose="books"
+        )
+
+        qu = self._db.query(ExternalIntegrationLink)
+        external_integration_links = qu.all()
+
+        eq_(len(external_integration_links), 2)
+        eq_(external_integration_links[0].other_integration_id, storage1.id)
+        eq_(external_integration_links[1].other_integration_id, storage2.id)
```

Comment:
In general you can't rely on objects coming out of a database query in a specific order. The simplest way to enforce this is to add `.order_by(ExternalIntegrationLink.other_integration_id)` to the `qu`. You can also do this by comparing two `set`s.